packages/editor/src/components/widgets/output/CardText.tsx

```diff
@@ -0,0 +1,59 @@
+import './CardText.css';
+import { IvyIcons } from '@axonivy/editor-icons';
+import IvyIcon from '../IvyIcon';
+import { splitMacroExpressions, splitNewLine } from '../../../utils/utils';
+import { FieldsetInputProps } from '../fieldset';
+import { ElementRef, forwardRef } from 'react';
+
+export type CardTextProps = Partial<FieldsetInputProps> & {
+  value?: string;
+};
+
+const Card = ({ part }: { part: string }) => {
+  const text = part.substring(3, part.length - 2).trim();
+  return (
+    <span className='card'>
+      <span className='card-icon'>
+        <IvyIcon icon={IvyIcons.Program} />
+      </span>
+      {text}
+    </span>
+  );
+};
+
+const CardOrText = ({ part }: { part: string }) => {
+  if (part.startsWith('<%=')) {
+    return <Card part={part} />;
+  }
+  return <span className='text'>{part}</span>;
+};
+
+const CardLine = ({ parts }: { parts: string[] }) => (
```

Comment:
Why not use an array direct here instead of wrapping it to an object?

Reply:
This is simply not supported by react. React always needs a single object as parameter (called props).

Let's expect that we don't have an object: `const CardLine = (parts: string[]) => (`
Then how would the call of this look like in JSX? `<CardLine ['abc'] />`

In HTML and also JSX every attribute needs to be named and react does this with the object `props`, but then you can call the component with the props in every order.

Reply:
Ok, thank for the education.